samza-api/src/main/java/org/apache/samza/context/TaskContext.java

```diff
@@ -0,0 +1,70 @@
+package org.apache.samza.context;
+
+import java.util.Set;
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.scheduling.Scheduler;
+import org.apache.samza.storage.kv.KeyValueStore;
+import org.apache.samza.system.SystemStreamPartition;
+import org.apache.samza.table.Table;
+
+
+/**
+ * Contains information at task granularity, provided by the Samza framework, to be used to instantiate an application
+ * at runtime.
+ * <p>
+ * Note that application-defined task-level context is accessible through {@link ApplicationDefinedTaskContext}.
+ */
+public interface TaskContext {
```

Comment:
Just curious.  This interface already exists in samza : `org.apache.samza.task.TaskContext`. Are we going to retain both the interfaces or merge both of them into one.

Reply:
Good question. This will be removed in the near future by a different PR. I added a comment to the javadoc to clarify.

Reply:
I am hoping removing the old one is also part of 1.0, since that will involve some changes to the current ContextManager interface in the ApplicationDescriptor API. I assume that ContextManager will be removed or replaced by this PR.